src/core/interruption.h

```diff
@@ -52,6 +52,21 @@ extern IGRAPH_THREAD_LOCAL igraph_interruption_handler_t *igraph_i_interruption_
         } \
     } while (0)
 
+/**
+ * \define IGRAPH_ALLOW_INTERRUPTION_LIMITED
+ *
+ * This is a variant of IGRAPH_ALLOW_INTERRUPTION() that checks for interruption
+ * only on every 'skips' call. It uses the 'iter' variable, which must be declared
+ * separately and initialized to 0, to keep track of the number of iterations.
```

Comment:
If we enforce the iteration tracker variable to be named as `iter` then there is no need to pass it as an argument to the macro. I'd suggest calling it `__iter` in the _macro_ and keeping the first argument of the macro so the user can choose whichever variable they want in the code to be used as an iteration variable.

Reply:
Sorry, I forgot the update the comment after I made the variable name a parameter.

Reply:
Okay, no worries. If you fix the comments, then we're done, feel free to merge then. Let me know when you think it's time for a new C patch release.

Reply:
@ntamas See chat about C patch release.